enumerables.rb

```diff
@@ -0,0 +1,107 @@
+# Enumerable methods
+
+module Enumerable
+  # my_each
+
+  def my_each
+    length.times { |i| yield(self[i]) }
+    self
+  end
+
+  # end of my_each
+
+  # my_each_with_index
+  def my_each_with_index
+    length.times { |i| yield(self[i], i) }
+    self
+  end
+
+  # end of my_each_with_index
+
+  # my_select
+
+  def my_select
+    arr = []
+    my_each { |i| arr.push(i) if yield(i) }
+    arr
+  end
+
+  # end of my_select
+
+  # my_all?
+
+  def my_all?
+    if_all = true
+    my_each do |i|
+      break if_all = false unless yield(i)
+    end
+    if_all
+  end
+
+  # end of my_all?
+
+  # my_any?
+
+  def my_any?
+    if_any = false
+    my_each do |i|
+      break if_any = true if yield(i)
+    end
+    if_any
+  end
+
+  # end of my_any?
+
+  # my_none?
+
+  def my_none?
+    if_none = true
+    my_each do |i|
+      break if_none = false if yield(i)
+    end
+    if_none
+  end
+
+  # end of my_none?
+
+  # my_count
+
+  def my_count(num)
+    count = 0
+    my_each { |i| count += 1 if num == i }
+    count
+  end
+
+  # end of my_count
+
+  # my_map
+
+  def my_map(num = nil)
+    arr = []
+    if block_given?
+      my_each { |i| arr.push(yield(i)) }
+    else
+      my_each { |i| arr.push(num.call(i)) }
+    end
+    arr
+  end
+
+  # end of my_map
+
+  # my_inject
+
+  def my_inject(acc = 0)
+    my_each { |i| acc = yield(acc, i) }
+    acc
+  end
```

Comment:
- [ ] should have working `my_inject` **(covered by automated tests)** identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

####  my_inject

- [ ] combines all elements of enum by applying a binary operation, specified by a block
```ruby
(1..5).my_inject(4) { |prod, n| prod * n } == (1..5).inject(4) { |prod, n| prod * n } => 480
```
- Yours is retrieving a NameError
![image](https://user-images.githubusercontent.com/52209067/81133797-f40d1180-8f0f-11ea-9af3-0649786eb884.png)

- Real inject
![image](https://user-images.githubusercontent.com/52209067/77698409-2ffaa180-6f76-11ea-91d3-9778aa560411.png)

- [ ] When a symbol is provided as an argument, my_inject should combine the elements of the enum using that symbol as a named method.
```ruby
[2, 5, 6].my_inject(:+) == [2, 5, 6].inject(:+) => 13
```
- Yours is retrieving a LocalJumpError
![image](https://user-images.githubusercontent.com/52209067/81133824-0f781c80-8f10-11ea-8203-82b493046739.png)

- Real inject
![image](https://user-images.githubusercontent.com/52209067/77698534-6a643e80-6f76-11ea-9d0e-ebb0f02da179.png)